### DISCUSSION TITLE
How to sort by date?

### DISCUSSION THREAD

**Discussion by OlegCinema** [prisma/prisma#6656] [Q&A] [answered]
Hello, i have User model. I wanna sort all users by date when they were created. Can I do this and how?

**Comment by m-abdelwahab** [comment, marked as answer]
Hey @OlegCinema  👋🏼 assuming you have a `createdAt` field in your User model, you can return all users sorted by using the [`orderBy API`](https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#sorting)

```js
const results = await prisma.user.findMany({
  orderBy: [
    {
      createdAt: "desc", // or pass "asc" to order ascendingly
    },
  ],
});
```

**Reply by OlegCinema** [reply]
Thanks! I have another question. How to return first 10 users from database?

**Reply by m-abdelwahab** [reply]
You would use `take`. The following snippet returns the first 10 users. If you're implementing pagination, then check out this [section](https://www.prisma.io/docs/concepts/components/prisma-client/pagination) in the docs

```js
 const results = await prisma.user.findMany({
    take: 10,
  });
```

**Reply by OlegCinema** [reply]
And if I want to find all users which was created from 10 January to 10 February, what should I do?

**Reply by OlegCinema** [reply]
> You would use `take`. The following snippet returns the first 10 users. If you're implementing pagination, then check out this [section](https://www.prisma.io/docs/concepts/components/prisma-client/pagination) in the docs
> 
> ```js
>  const results = await prisma.user.findMany({
>     take: 10,
>   });
> ```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
